storm-core/src/clj/org/apache/storm/internal/thrift.clj

```diff
@@ -0,0 +1,291 @@
+;; Licensed to the Apache Software Foundation (ASF) under one
+;; or more contributor license agreements.  See the NOTICE file
+;; distributed with this work for additional information
+;; regarding copyright ownership.  The ASF licenses this file
+;; to you under the Apache License, Version 2.0 (the
+;; "License"); you may not use this file except in compliance
+;; with the License.  You may obtain a copy of the License at
+;;
+;; http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+
+(ns org.apache.storm.internal.thrift
+  (:import [java.util HashMap]
+           [java.io Serializable]
+           [org.apache.storm.generated NodeInfo Assignment])
+  (:import [org.apache.storm.generated JavaObject Grouping Nimbus StormTopology
+            StormTopology$_Fields Bolt Nimbus$Client Nimbus$Iface
+            ComponentCommon Grouping$_Fields SpoutSpec NullStruct StreamInfo
+            GlobalStreamId ComponentObject ComponentObject$_Fields
+            ShellComponent SupervisorInfo])
+  (:import [org.apache.storm.utils Utils NimbusClient ConfigUtils])
+  (:import [org.apache.storm Constants])
+  (:import [org.apache.storm.security.auth ReqContext])
+  (:import [org.apache.storm.grouping CustomStreamGrouping])
+  (:import [org.apache.storm.topology TopologyBuilder])
+  (:import [org.apache.storm.clojure RichShellBolt RichShellSpout])
+  (:import [org.apache.thrift.transport TTransport])
+  (:use [org.apache.storm util config log zookeeper]))
+
+(defn instantiate-java-object
+  [^JavaObject obj]
+  (let [name (symbol (.get_full_class_name obj))
+        args (map (memfn getFieldValue) (.get_args_list obj))]
+    (eval `(new ~name ~@args))))
```

Comment:
Can you please delete the functions that are not needed?  I know you want to remove them after the review, but it actually makes the review simpler for me to have them gone, so I can see what the final code really will look like.